protos/flyteidl/service/signal.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+package flyteidl.service;
+
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/service";
+
+import "google/api/annotations.proto";
+import "flyteidl/admin/signal.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+
+// SignalService defines an RPC Service that may create, update, and retrieve signal(s).
+service SignalService {
+  // Fetches or creates a :ref:`ref_flyteidl.admin.Signal`.
+  rpc GetOrCreateSignal (flyteidl.admin.SignalGetOrCreateRequest) returns (flyteidl.admin.Signal) {
+    // Purposfully left out an HTTP API for this RPC call. This is meant to idempotently retrieve
```

Comment:
hopefully one day we'll have authz!

Reply:
Does FlyteAdmin have a scheme implemented already? Happy to add something existing. I'm sure starting from the ground up would be a bit out of scope for this work.